### PR TITLE
Edit/Update Feature Closes #113

### LINKED ISSUE #113: Add edit feature for existing projects
Currently, there is no way to edit existing peojects. Add an edit feature including access control checks.

### PR DESCRIPTION
**Pull request by leosa83**
Using the quickForm package.

### CODE REVIEW

File: projects/update/client/updateProject.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+Template.updateProject.rendered = function() {
+  // Get reference to template instance
+  let instance = this;
```
Comment (brylie) [inline projects/update/client/updateProject.js]: It is safe to use `const` here, since `instance` will not be re-assigned.

# Reference
- MDN [const - JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const)

File: projects/update/client/updateProject.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          }
+        }
+      });
+    }
+  });
+};
+
+Template.updateProject.helpers({
+    //Using the Flowrouter package to get the route param projects/:projectId ; is set up in the Template.updateProject.onCreated callback
+    getProject: function () {
```
Comment (brylie) [inline projects/update/client/updateProject.js]: Lets shorten this to `projects` for brevity. This would also be consistent with our other template helpers, which tend to be nouns - as opposed to the current verb phrase `get project`.

File: projects/update/client/updateProject.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  existingTags: function () {
+    // Get all existing tags
+    let tags = Tags.find().fetch();
+
+    return tags;
+  }
+});
+
+Template.updateProject.onCreated(function(){
+  var self = this;
```
Comment (brylie) [inline projects/update/client/updateProject.js]: Remove this line, since there is `let instance = this` on the following statement.

File: projects/update/client/updateProject.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  // Get reference to template instance
+  let instance = this;
+
+  // Subscribe to all tags, for tag auto-complete field
+  instance.subscribe("allTags");
+  
+  self.autorun(function() {
```
Comment (brylie) [inline projects/update/client/updateProject.js]: Use `instance` instead of `self`.

File: projects/update/client/updateProject.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    var projectId = FlowRouter.getParam('projectId');
```
Comment (brylie) [inline projects/update/client/updateProject.js]: Use `let` or `const` instead of var.

File: projects/update/client/updateProject.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    self.subscribe('singleProject', projectId);
```
Comment (brylie) [inline projects/update/client/updateProject.js]: Use `instance` instead of `self`, for consistency with other project files.

File: projects/update/client/updateProject.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    return tags;
+  }
+});
+
+Template.updateProject.onCreated(function(){
+  var self = this;
+
+  // Get reference to template instance
+  let instance = this;
```
Comment (brylie) [inline projects/update/client/updateProject.js]: It is safe to use `const` here, since `instance` will not be re-assigned.

File: projects/update/client/updateProject.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    }
+  });
+};
+
+Template.updateProject.helpers({
+    //Using the Flowrouter package to get the route param projects/:projectId ; is set up in the Template.updateProject.onCreated callback
+    getProject: function () {
+      return Projects.findOne({ _id: FlowRouter.getParam('projectId')});
+    },
+  existingTags: function () {
```
Comment (brylie) [inline projects/update/client/updateProject.js]: Where is this template helper used? Remove if unused.

File: projects/update/client/updateProject.html [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<template name="updateProject">
+    <div class="page-header">
+      <h2>Update Project</h2>
+    </div>
+    
+    {{> quickForm collection="Projects" doc=getProject id="updateProjectForm" type="update"}}
```
Comment (brylie) [inline projects/update/client/updateProject.html]: Shorten `getProject` to `project` for brevity and convention. I.e. we are generally using nouns for template helpers that return an object, as opposed to a verb phrase ('get project').

File: projects/update/client/updateProject.js (hunk 8 above)
Comment (leosa83) [inline projects/update/client/updateProject.js]: Ath the moment it is unused since we are not using the autoForm package as in the addProject template

Comment (brylie) [inline projects/update/client/updateProject.js]: OK, cool. Please remove then :-)

**Comment by leosa83** [comment]
Now i think its ready for merging.

File: projects/update/client/updateProject.html (hunk 9 above)
Comment (brylie) [inline projects/update/client/updateProject.html]: Remember to update the `doc=project` property, so that it uses the `project` template helper instead of `getProject`.

Comment (leosa83) [inline projects/update/client/updateProject.html]: Will do

**Comment by leosa83** [comment]
This was reviewed and can now be merged.
